lib/rules/object-property-newline.js

```diff
@@ -34,20 +40,58 @@ module.exports = {
 
     create(context) {
         const allowSameLine = context.options[0] && Boolean(context.options[0].allowMultiplePropertiesPerLine);
-        const errorMessage = allowSameLine
+        const allowOpenBracket = context.options[0] && Boolean(context.options[0].treatComputedPropertiesLikeJSCS);
+        const denyCommaFirst = context.options[0] && Boolean(context.options[0].noCommaFirst);
+        let errorMessage = allowSameLine
             ? "Object properties must go on a new line if they aren't all on the same line."
             : "Object properties must go on a new line.";
 
+        if (allowOpenBracket) {
+            errorMessage += " The opening bracket of a computed property name may end a line on which another property appears.";
+        }
+
+        if (denyCommaFirst) {
+            errorMessage += " The comma delimiting two properties may not share a line with any of the second property.";
+        }
+
         const sourceCode = context.getSourceCode();
 
+        /**
+         * Returns whether a property begins with a bracket that ends its line.
+         * @param {token} myToken0 Current property’s first token.
```

Comment:
Nit: I think `firstToken` or `propertyFirstToken` would be a better variable name.